torch/distributed/device_mesh.py

```diff
@@ -553,6 +554,9 @@ def init_device_mesh(
                     f"Found len(mesh_dim_names): {len(mesh_dim_names)} and len(mesh_shape):{len(mesh_shape)}.",
                 )
 
+        # e.g., strip the ':0' from 'cuda:0'
+        if device_type:
+            device_type = device_type.split(":")[0]
```

Comment:
Any other better way to implement this?

Reply:
My personal opinion is that we should disallow passing `"cuda:0"`.

I would prefer if the user/trainer code maintained a `torch.device` object like `device = torch.device("cuda:0")` and passed `init_device_mesh(device.type, ...)` instead of maintaining the device as a string. @wconstab Would using a device object instead of a string be okay in the trainer code?

Reply:
Based on the discussion, it looks like we need to add some validation for this device_type passing in from the user. 

In the docstring, we currently list `device_type (str): The device type of the mesh. Currently supports: "cpu", "cuda/cuda-like"` but seems we need to validate this during DeviceMesh init, something like `device_type` should be a str with no digit? Does that sound good to folks?

Reply:
Yeah, we should  clarify what the accepted values are for `device_type`. For example:

- If you hold a `torch.device` object, it's recommended to pass in `device.type`.
- If you are using a string, then it should not contain a gpu index or you will get an exception.

Following code assumes that valid device types all consist of letters. Any suggestions are welcome.

```suggestion
        if device_type and not device_type.isalpha():
            raise RuntimeError(
                f"Device type with GPU index is not supported but got {device_type}.",
                "If you hold a 'torch.device' object, it's recommended to pass in 'device.type'.",
            )
```

Reply:
Thanks for the updates. This looks good to me.